### ISSUE TITLE
PPOCRv5 supports line recognition (Size and number of characters don't match in the config)

### ISSUE THREAD

**Issue by bely66** [PaddlePaddle/PaddleOCR#16476]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

Hi Everyone,

Thanks for the great work, the technical report is amazing, and the results are really good.

When i'm trying to use it for my specific data i'm facing some challenges:
1. In the config of ppocrv5-rec-server:
a.  the maximum chars are 25 (5-7 words maximum
b. The image size is [3, 48, 320], the ratio is completely unsuitable for training recognition line images

How can i get the same results in the technical report with this configuration what do i need to change and would the model work this way?

### 🏃‍♂️ Environment (运行环境)

Python 3.12
OS Ubuntu 
PaddleOCR Latest

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

Global:
  model_name: PP-OCRv5_server_rec # To use static model for inference.
  debug: false
  use_gpu: true
  epoch_num: 75
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/PP-OCRv5_server_rec
  save_epoch_step: 1
  eval_batch_step: [0, 2000]
  cal_metric_during_train: true
  calc_epoch_interval: 1
  pretrained_model: 
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ./ppocr/utils/dict/ppocrv5_dict.txt
  max_text_length: &max_text_length 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv5.txt
  d2s_train_image_shape: [3, 48, 320]

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
    warmup_epoch: 1
  regularizer:
    name: L2
    factor: 3.0e-05

Architecture:
  model_type: rec
  algorithm: SVTR_HGNet
  Transform:
  Backbone:
    name: PPHGNetV2_B4
    text_rec: True
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
          Head:
            fc_decay: 0.00001
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: ./train_data/
    ext_op_transform_idx: 1
    label_file_list:
    - ./train_data/train_list.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[320, 32], [320, 48], [320, 64]]
    first_bs: &bs 128
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:
    shuffle: true
    batch_size_per_card: *bs
    drop_last: true
    num_workers: 16
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data
    label_file_list:
    - ./train_data/val_list.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 4

**Comment by liuhongen1234567** [comment]
Hello, regarding issue 1, in ppocrv5-rec-server, there are two heads during training: CTC Head and NRTR Head. The "maximum chars" setting affects the NRTR Head, while the CTC Head does not have a character length limit. During inference, the CTC Head is used, so long text can still be output normally. As for issue 2, our experiments have shown that [3, 48, 320] is just an aspect ratio, and in fact, 320 pixels can already carry a lot of information, so it will not affect the normal training of the model. This configuration has also been used in previous generations of the PP-OCR series models.

**Comment by binhere** [comment]
Hello @liuhongen1234567 I am seeking the answer for this long text problem. Thank you for your answer, it is very helpful for me when training on a Vietnamese sentence per sample recognizer. However, I have a couple of questions.

> in ppocrv5-rec-server, there are two heads during training: CTC Head and NRTR Head. The "maximum chars" setting affects the NRTR Head, while the CTC Head does not have a character length limit. During inference, the CTC Head is used, so long text can still be output normally

As you said, max_text_length only affects to NRTR Head when training, and does not affect CTC Head when inference without length limit. Because my dataset has sentences that exceed 25 characters, I am stucked with following questions:

1. What do I need to adjust the configuration yml file properly (such as Global.d2s_train_image_shape)? 
2. If keeping length to 25, what happens with my long text/senetence? The model will use them or discard them?
3. How these samples (affect the NRTR Head) affects to model performance? 
4. As you said [3, 48, 320] is just an aspect ratio, so my image of sentence can vary (for example, my image could be loaded in [3, 60, 400] in pixels, that means it has the same default aspect ratio 48:320 = 60:400, right?) In fact, one typical sample in my dataset has shape 1444x64 , most of image heights are 64 and widths vary

**Comment by liuhongen1234567** [comment]
Hello, for the first issue, you can adjust max_text_length to increase the text length during model training, and use scales: [[320, 32], [320, 48], [320, 64]] to adjust the image scaling ratio during model training. However, adjusting max_text_length may result in some pre-trained weights not being loaded.

For the second issue, keeping length=25 will cause the model to discard this data during training.

For the third issue, since these excessively long texts are discarded during training, the model's performance on such long texts will be slightly worse compared to including this data in the training.

For the fourth issue, resizing images to a fixed size is a common practice in OCR training. The model can implicitly learn to utilize the various aspect ratios of input images. The current configuration is optimized based on practical experience with the PP-OCR series for most scenarios, but it may not necessarily be suitable for your dataset. You can experiment by modifying the use scales: [[320, 32], [320, 48], [320, 64]] to select different sizes.

**Comment by TingquanGao** [comment]
The issue has no response for a long time and will be closed. You can reopen or new another issue if are still confused.

---
_From Bot_
